Products/ZenPackAdapter/zobject.py

```diff
@@ -595,12 +595,14 @@ def device(self):
         return self
 
     def getDeviceComponents(self, collector=None):
+        components = []
         for component in self.getSubComponents():
             if collector is not None:
                 if collector in getattr(component, 'collectors', []):
-                    yield component
+                    components.append(component)
```

Comment:
not that I imagine it matters much, but why the change here?  this is slightly less efficient..

Reply:
when running zendmd on resmgr, a list with the components were provided
> dev1.getDeviceComponents()
[ compont_type_name_id_1,
compont_type_name_id_2,
etc,
] 

When running dmd on ZPA, the list is not generated, instead it's handle is returned
> dev1.getDeviceComponents()
(generator_object)

Reply:
Sure, that's typical of generators, but it's functionally equivalent (you can iterate on it, same as a list, or turn it into one  `list(dev1.getDeviceComponents())`for debugging)